src/averageAges.js

```diff
@@ -55,7 +77,16 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  return people
+    .filter(person => onlyWithSon
+      ? person.sex === 'm'
+      : person)
+    .filter(child => people.some(mother => mother.name === child.mother))
+    .reduce((prev, x, index, array) => {
+      return (index < array.length - 1)
+        ? prev + x.born - people.find(mother => mother.name === x.mother).born
```

Comment:
you can move `people.find(mother => mother.name === x.mother).born` to the variable to make code easier to read